dom/events/scrolling/wheel-event-groups.html

```diff
@@ -0,0 +1,142 @@
+    </div>
+    <div id="secondSpacer" class="spacer" style="background: blue">
+    </div>
+  </div>
+</div>
+</body>
+
+<script>
+
+let scrollable_wheel_event = false;
+scrollableDiv.addEventListener("wheel", (e) => {
+  e.stopPropagation();
+  scrollable_wheel_event = true;
+});
+
+let inner_wheel_event = false;
+innerDiv.addEventListener("wheel", (e) => {
+  e.stopPropagation();
+  inner_wheel_event = true;
+});
+
+function runTest() {
+  promise_test(async (t) => {
+    // Start a wheel scroll above the scrollable child element, and
+    // scroll over the child element to ensure that wheel events
+    // part of this transaction have the root element as their target.
+    await new test_driver.Actions()
+      .addWheel("wheel1")
+      .scroll(40, 2, 0, 20)
+      .scroll(40, 2, 0, 100)
+      .pause(1)
+      .scroll(40, 2, 0, 50)
+      .pause(500)
+      .send();
+
+    assert_equals(inner_wheel_event, false, "Wheel event was not fired to child");
+    assert_equals(scrollable_wheel_event, false, "Wheel event was not fired to scrollable child");
+  }, 'Wheel events are grouped and are not sent to a child element');
+
+  promise_test(async (t) => {
+    document.scrollingElement.scrollTo({top: 0, lef: 0});
+    scrollable_wheel_event = false;
+    inner_wheel_event = false;
+
+    await waitForCompositorCommit();
+
+    // Scroll in the child and ensure that wheel events in the group are not
+    // directly fired to the scrollable element.
+    await new test_driver.Actions()
+      .addWheel("wheel1")
+      .scroll(40, 2, 0, 20, {origin: innerDiv})
+      .scroll(40, 2, 0, 100, {origin: innerDiv})
+      .pause(1)
+      .scroll(40, 2, 0, 50, {origin: innerDiv})
+      .pause(500)
+      .send();
+
+    assert_equals(inner_wheel_event, true, "Wheel event was fired to inner div");
+    assert_equals(scrollable_wheel_event, false, "Wheel event was not fired to scrollable child");
+  }, 'Wheel events are not bound to a scroll frame');
+
+  promise_test(async (t) => {
+    document.scrollingElement.scrollTo({top: 0, left: 0});
+    scrollableDiv.scrollTo({top: 0, left: 0});
+    scrollable_wheel_event = false;
+    inner_wheel_event = false;
+
+    await waitForCompositorCommit();
+
+    let first_spacer_wheel_event = false;
+    firstSpacer.addEventListener("wheel", (e) => {
+      e.stopPropagation();
+      first_spacer_wheel_event = true;
+    });
+
+    let second_spacer_wheel_event = false;
+    secondSpacer.addEventListener("wheel", (e) => {
+      e.stopPropagation();
+      second_spacer_wheel_event = true;
+    });
+
+    // Scroll past the boundary of the original element to ensure all events in
+    // group have the same target.
+    await new test_driver.Actions()
+      .addWheel("wheel1")
+      .scroll(40, 2, 0, 20, {origin: firstSpacer})
+      .scroll(40, 2, 0, 100, {origin: firstSpacer})
+      .pause(1)
+      .scroll(40, 2, 0, 50, {origin: firstSpacer})
```

Comment:
In my understanding, `firstSpacer` is now scrolled out after the previous `.scroll` call. Does this call emulate user input in any browsers?

Reply:
Yeah, this emulates what happens for several small wheel scrolls that start on `firstSpacer` that happen in quick succession. I'll add a test that doesn't add an origin for additional coverage.

Reply:
Added a test that does what this test did without an element as an origin. It intermittently fails on WebKit, but the behavior is always the same for a manual test.